autoload/SpaceVim/lsp.vim

```diff
@@ -8,7 +8,38 @@
 
 scriptencoding utf-8
 
-if has('nvim')
+if g:spacevim_autocomplete_method ==# 'coc'
+  " use coc.nvim
+  let s:coc_language_servers = {}
+  function! SpaceVim#lsp#reg_server(ft, cmds) abort
+    if has_key(s:coc_language_servers, a:cmds[0])
+      call add(s:coc_language_servers[a:cmds[0]].filetypes, a:ft)
+    else
+      let s:coc_language_servers[a:cmds[0]] = {
+            \'command': a:cmds[0],
+            \'args': a:cmds[1:],
+            \'filetypes': [a:ft]
+            \}
+    endif
+    autocmd! User CocSetup :call coc#config("languageserver", s:coc_language_servers)
```

Comment:
`SpaceVim#lsp#reg_server()` will be called in every language layer, when this function is called, an autocmd is definded, should we move this autocmd to coc's config.
```
config/plugins/coc.vim
```

Reply:
Note that with the `!` at the end of the `autocmd`, the `autocmd` will be removed and re-registered at each execution. Still, this may be some overhead, but IMHO it's quite unnoticeable.

However, if you think that it's preferable to call `coc#config` in `config/plugins/coc.vim`, I've totally fine with it.
In that case, do you know a way to share `s:coc_language_servers` with `config/plugins/coc.vim`? I'm not familiar with Vimscript and on Google I didn't find anything.
A quick solution would be to make `coc_language_servers` global (so `g:coc_language_servers`), so as to make it accessible to `config/plugins/coc.vim`, but I don't know whether it's fine for you to pollute the global namespace.